lighthouse-core/lib/page-functions.js

```diff
@@ -218,6 +218,63 @@ function getNodeSelector(node) {
   return parts.join(' > ');
 }
 
+/**
+ * @param {HTMLElement} element
+ * @param {String} attr
+ * @return {String}
+ */
+/* istanbul ignore next */
+function getStyleAttrValue(element, attr) {
+  // Check style before computedStyle as computedStyle is expensive.
+  return element.style[attr] || window.getComputedStyle(element)[attr];
+}
+
+/**
+ * @param {HTMLElement} element
+ * @return {Boolean}
+ */
+/* istanbul ignore next */
+function hasScrollableAncestor(element) {
+  let currentEl = element.parentElement;
+  while (currentEl) {
+    if (currentEl.scrollHeight > currentEl.clientHeight) {
+      const yScroll = getStyleAttrValue(currentEl, 'overflowY');
+      if (yScroll) {
+        return true;
+      }
+    }
+    currentEl = currentEl.parentElement;
+  }
+  return false;
+}
+
+/**
+ * @param {?HTMLElement} element
+ * @return {Boolean}
```

Comment:
Any reason not to use primitive types (`string` and `boolean` vs `String` and `Boolean`)? here and elsewhere